modules/flowable-engine/src/main/java/org/flowable/engine/impl/bpmn/helper/ClassDelegate.java

```diff
@@ -137,9 +137,13 @@ protected ExecutionListener getExecutionListenerInstance() {
         if (delegateInstance instanceof ExecutionListener) {
             return (ExecutionListener) delegateInstance;
         } else if (delegateInstance instanceof JavaDelegate) {
-            return new ServiceTaskJavaDelegateActivityBehavior((JavaDelegate) delegateInstance, triggerable, skipExpression);
+            return new ServiceTaskJavaDelegateActivityBehavior(
```

Comment:
I think that it makes more sense to do 
```
return execution -> ((JavaDelegate) delegateInstance).execute(execution);
```

instead of adapting the call to the `ServiceTaskJavaDelegateActivityBehavior` for this. This is more or less similar to what is done for a delegate expression. Perhaps we can even have a new implementation class for this. Similar like `DelegateExpressionExecutionListener`. 

Another alternative would be to do exactly like it is done in the `DelegateExpressionExecutionListener#notify` and handle the exception throwing there.

Reply:
Thanks for the quick look! Yh, I was a bit surprised that the ExecutionListener is being handled by an *ActivityBehavior too during the analysis. Also thought that It'd be better to have this handled by a more specific implementation. 
But then the lack of experience just kept me to keep it small and wait for a feedback :-)

Reply:
Adapted the implementation to use a new `DelegateExecutionListener`